tests/unit/test_bpm_calibration.py

```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+import numpy as np
+import pandas as pd
+import pathlib
+import pytest
+
+import tfs
+from generic_parser.dict_parser import ArgumentError
+from pylhc import BPM_calibration as calibration
+
+INPUTS_DIR = Path(__file__).parent.parent / 'inputs' / 'calibration'
+MEASUREMENTS_BETA = INPUTS_DIR / 'measurements' / 'for_beta'
+MEASUREMENTS_DISPERSION = INPUTS_DIR / 'measurements' / 'for_dispersion'
+EXPECTED_OUTPUT = INPUTS_DIR / 'output'
+
+
+def test_calibration_same_betabeat(tmp_path):
+    factors = calibration.main(input_path=MEASUREMENTS_BETA,
+                               output_path=tmp_path,
+                               ips=[1,4,5])
+
+    # Let's open the tfs files we just created
+    x_tfs = tfs.read(tmp_path / 'calibration_beta_x.tfs')
+    y_tfs = tfs.read(tmp_path / 'calibration_beta_y.tfs')
+    
+    # And the ones created by BetaBeat.src for the same measurements
+    expected_x_tfs = tfs.read(EXPECTED_OUTPUT / 'calibration_beta_x.tfs')
+    expected_y_tfs = tfs.read(EXPECTED_OUTPUT / 'calibration_beta_y.tfs')
+
+    # Check all the BPMs are indeed the same 
+    assert x_tfs['NAME'].equals(expected_x_tfs['NAME'])
+    assert y_tfs['NAME'].equals(expected_y_tfs['NAME'])
+
+    # Drop the NAME column for float comparison
+    x_tfs = x_tfs.drop(['NAME'], axis=1)
+    y_tfs = y_tfs.drop(['NAME'], axis=1)
+    expected_y_tfs = expected_y_tfs.drop(['NAME'], axis=1)
+    expected_x_tfs = expected_x_tfs.drop(['NAME'], axis=1)
+
+    # BetaBeat's tfs implementation is a bit different, we don't have the
+    # same integer precision
+    precision = 1e-14
+
+    assert np.allclose(x_tfs, expected_x_tfs, atol=precision)
+    assert np.allclose(y_tfs, expected_y_tfs, atol=precision)
+
```

Comment:
Pandas provides a [function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.testing.assert_frame_equal.html) to check if df are close enough .. might allow to shorten tests a bit

Reply:
It indeed works fine, it also now checks the Index which I forgot before